clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -186,6 +187,7 @@ public ConsumerCoordinator(GroupRebalanceConfig rebalanceConfig,
         this.completedOffsetCommits = new ConcurrentLinkedQueue<>();
         this.sensors = new ConsumerCoordinatorMetrics(metrics, metricGrpPrefix);
         this.interceptors = interceptors;
+        this.inFlightAsyncCommits = new AtomicInteger();
         this.pendingAsyncCommits = new AtomicInteger();
```

Comment:
@philipnee @kirktrue @erikvanoosten I was looking at how we use `pendingAsyncCommits` in `close()`. My understanding is that we use it to wait in close until all the in-flight commit requests are done to ensure that their callbacks are called. Knowing this, isn't it weird that we decrement `pendingAsyncCommits` when we resolve the coordinator and not when the response of the offset commit is received? I may be missing something but this is not clear to me.

I was looking into this because I was hoping that we could consolidate `inFlightAsyncCommits` and `pendingAsyncCommits`.

Reply:
Okay. I understand this a bit better now. `pendingAsyncCommits` is really only meant to track inflight coordinator lookups. Then, in `close`, we wait on those to complete before calling `close` of the super class which waits on all the inflight requests to complete.

Reply:
Starting from commit https://github.com/apache/kafka/pull/13678/commits/5f8bfc42e9791a97c01cecef334fccc88dd547f2 this is much clearer because only a single method updates `inFlightAsyncCommits`.